partner_contact_birthdate/models/res_partner.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (C) 2014-2015  Grupo ESOC <www.grupoesoc.es>
+# © 2017-Apertoso N.V. (<http://www.apertoso.be>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+import logging
+
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    """Partner with birth date in date format."""
+    _inherit = "res.partner"
+
+    # New birthdate field in date format
+    birthdate_date = fields.Date("Birthdate")
+
+    # Make the old Char field to reflect the new Date field
+    birthdate = fields.Char(
+        compute="_birthdate_compute",
+        inverse="_birthdate_inverse",
+        store=True)
+
+    @api.one
```

Comment:
Why are you using api.one and not api.multi?

Reply:
We could add `oca-decorators` to the python dependencies, then switch to `oca.decorators.foreach`